src/movies.js

```diff
@@ -1,15 +1,2102 @@
+    director: "Wim Wenders",
+    duration: "2h 25min",
+    genre: ["Drama"],
+    rate: 8.1
+  },
+  {
+    title: "The Help",
+    year: 2011,
+    director: "Tate Taylor",
+    duration: "2h 26min",
+    genre: ["Drama"],
+    rate: 8.1
+  },
+  {
+    title: "Faa yeung nin wa",
+    year: 2000,
+    director: "Kar-Wai Wong",
+    duration: "1h 38min",
+    genre: ["Drama", "Romance"],
+    rate: 8.1
+  },
+  {
+    title: "Sholay",
+    year: 1975,
+    director: "Ramesh Sippy",
+    duration: "3h 18min",
+    genre: ["Action", "Adventure", "Comedy", "Drama", "Musical", "Thriller"],
+    rate: 8.2
+  },
+  {
+    title: "Beauty and the Beast",
+    year: 1991,
+    director: "Gary Trousdale",
+    duration: "1h 24min",
+    genre: ["Animation", "Family", "Fantasy", "Musical", "Romance"],
+    rate: 8
+  },
+  {
+    title: "La battaglia di Algeri",
+    year: 1966,
+    director: "Gillo Pontecorvo",
+    duration: "2h 1min",
+    genre: ["Drama", "War"],
+    rate: 8.1
+  },
+  {
+    title: "Ah-ga-ssi",
+    year: 2016,
+    director: "Chan-wook Park",
+    duration: "2h 24min",
+    genre: ["Crime", "Drama", "Mystery", "Romance", "Thriller"],
+    rate: 8.1
+  },
+  {
+    title: "Pirates of the Caribbean: The Curse of the Black Pearl",
+    year: 2003,
+    director: "Gore Verbinski",
+    duration: "2h 23min",
+    genre: ["Action", "Adventure", "Fantasy"],
+    rate: 8
+  },
+  {
+    title: "PK",
+    year: 2014,
+    director: "Rajkumar Hirani",
+    duration: "2h 33min",
+    genre: ["Comedy", "Drama", "Fantasy", "Sci-Fi"],
+    rate: 8.2
+  },
+  {
+    title: "Dog Day Afternoon",
+    year: 1975,
+    director: "Sidney Lumet",
+    duration: "2h 5min",
+    genre: ["Biography", "Crime", "Drama", "Thriller"],
+    rate: 8
+  },
+  {
+    title: "Dead Poets Society",
+    year: 1989,
+    director: "Peter Weir",
+    duration: "2h 8min",
+    genre: ["Comedy", "Drama"],
+    rate: 8
+  }
+];
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(movies) {
+  const newArray = [...movies];
+  newArray.sort(function(objA, objB) {
+    if (objA.year < objB.year) {
+      return -1;
+    }
+    if (objA.year > objB.year) {
+      return 1;
+    }
+    if (objA.year === objB.year) {
+      if (objA.title < objB.title) {
+        return -1;
+      }
+      if (objA.title > objB.title) {
+        return 1;
+      }
+      if (objA.title === objB.title) {
+        return 0;
+      }
+      return 0;
+    }
+  });
+  return newArray;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(movies) {
+  const newArray = movies.filter(function(genereObj, i, movies) {
+    if (genereObj.genre == "Drama") {
+      return true;
+    }
+  });
+  console.log("newArray", newArray);
+}
+howManyMovies(movies);
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(movies) {
```

Comment:
Remember both sort and splice mutate the original array (even when passed as a parameter). The instructions asked you not to do that, so you would need to spread the array into a new variable. The same way you did in the first step of iteration 1.
Also, return the result instead of console.log